app/models/king.rb

```diff
@@ -1,2 +1,12 @@
 class King < Piece
+
+  def valid_move?(destination_x, destination_y)
+    super
+    if (destination_x - self.x_position).abs > 1 || (destination_y - self.y_position).abs > 1
```

Comment:
Trick with conditionals, removing the `if` makes them a boolean expression. You can switch your logic around a bit in order to return the expression, instead of needing the nested return switch.